streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -2474,7 +2474,9 @@ void process(final ProcessorSupplier<? super K, ? super V> processorSupplier,
      * For each {@code KStream} record that finds a corresponding record in {@link GlobalKTable} the provided
      * {@link ValueJoiner} will be called to compute a value (with arbitrary type) for the result record.
      * The key of the result record is the same as the key of this {@code KStream}.
-     * If an {@code KStream} input record key or value is {@code null} the record will not be included in the join
+     * If {@code keyValueMapper} returns {@code null} implying no match exists, no output record will be added to the
+     * resulting {@code KStream}.
+     * If a {@code KStream} input record key or value is {@code null} the record will not be included in the join
```

Comment:
Nit: note that for this case, `KeyValueMapper` won't be called in the first place? (same below)?

Maybe we should restructure the paragraph an move this sentence up?

Reply:
Agreed. Reads better the other way round.